CoreWidgetProvider/Helpers/ChartHelper.cs

```diff
@@ -1,138 +1,245 @@
+                stop1Style = NetBrushStop1Style;
+                stop2Style = NetBrushStop2Style;
+                break;
+            case ChartType.CPU:
+            default:
+                stop1Style = CPUBrushStop1Style;
+                stop2Style = CPUBrushStop2Style;
+                break;
+        }
+
+        var stop1 = new XElement(StopElement);
+        stop1.SetAttributeValue(OffsetAttr, "0%");
+        stop1.SetAttributeValue(StyleAttr, stop1Style);
+
+        var stop2 = new XElement(StopElement);
+        stop2.SetAttributeValue(OffsetAttr, "95%");
+        stop2.SetAttributeValue(StyleAttr, stop2Style);
+
+        gradientElement.Add(stop1);
+        gradientElement.Add(stop2);
+        defsElement.Add(gradientElement);
+
+        return defsElement;
+    }
+
+    private static XElement CreateBorderBox(int height, int width)
+    {
+        var boxElement = new XElement(RectElement);
+        boxElement.SetAttributeValue(HeightAttr, height);
+        boxElement.SetAttributeValue(WidthAttr, width);
+        boxElement.SetAttributeValue(StyleAttr, LightGrayBoxStyle);
+        return boxElement;
+    }
+
+    private static string GetLineStyle(ChartType type)
+    {
+        var lineStyle = type switch
+        {
+            ChartType.CPU => CPULineStyle,
+            ChartType.GPU => GPULineStyle,
+            ChartType.Mem => MemLineStyle,
+            ChartType.Net => NetLineStyle,
+            _ => CPULineStyle,
+        };
+
+        return lineStyle;
+    }
+
+    private static StringBuilder TransformPointsToLine(List<float> chartValues, out int startX, out int finalX)
+    {
+        var points = new StringBuilder();
+
+        // The X value where the graph starts must be adjusted so that the graph is right-aligned.
+        // The max available width of the widget is 268. Since there is a 1 px border around the chart, the width of the chart's line must be <=266.
+        // To create a chart of exactly the right size, we'll have 30 points with 9 pixels in between:
+        // index 0            1                      2 - 262                          263
+        // 1 px left border + 1 px for first point + 29 segments * 9 px per segment + 1 px right border = 264 pixels total in width.
+        var pxBtwnPoints = 9;
+
+        // When the chart doesn't have all points yet, move the chart over to the right by increasing the starting X coordinate.
+        // For a chart with only 1 point, the svg will not render a polyline.
+        // For a chart with 2 points, starting X coordinate ==  1 + (30 -  2) * 9 == 1 + 28 * 9 == 1 + 252 == 253
+        // For a chart with 30 points, starting X coordinate == 1 + (30 - 30) * 9 == 1 +  0 * 9 == 1 +   0 ==   1
+        startX = 1 + ((MaxChartValues - chartValues.Count) * pxBtwnPoints);
+        finalX = startX;
+        foreach (var origY in chartValues)
+        {
+            var finalY = ChartHeight - 1 - origY;
+            points.Append(finalX + "," + finalY + " ");

```

Comment:
nit, could do string interpolation (here and other places) like
```suggestion
            points.Append($"{finalX},{finalY}");
```